cwms-data-api/src/main/java/cwms/cda/api/BlobController.java

```diff
@@ -138,50 +156,118 @@ public void getAll(Context ctx) {
             queryParams = {
                     @OpenApiParam(name = OFFICE, description = "Specifies the owning office."),
             },
-            tags = {"Blob"}
+            tags = {TAG}
     )
     @Override
-    public void getOne(Context ctx, String blobId) {
+    public void getOne(@NotNull Context ctx, @NotNull String blobId) {
 
-        try (final Timer.Context timeContext = markAndTime(GET_ONE);        ) {
-                DSLContext dsl = getDslContext(ctx);
+        try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+            DSLContext dsl = getDslContext(ctx);
             BlobDao dao = new BlobDao(dsl);
             String officeQP = ctx.queryParam(OFFICE);
             Optional<String> office = Optional.ofNullable(officeQP);
-            Optional<Blob> optAc = dao.getByUniqueName(blobId, office);
-
-            if (optAc.isPresent()) {
-                Blob blob = optAc.get();
 
-                ctx.contentType(blob.getMediaTypeId());
-                byte[] value = blob.getValue();
-                ctx.result(value);
-
-                requestResultSize.update(value.length);
+            Consumer<Triple<InputStream, Long, String>> tripleConsumer = triple -> {
```

Comment:
Neat trick.

Reply:
I don't love it but I'm hoping we can try it for now.   It puts a kotlin class in the interface signature.  But the code is only going to work with Javalin which makes heavy use of kotlin so that seemed the fastest solution.  I didn't see a similar Java class sitting around.  Maybe we can create our own ThreeTuple class or a TriConsumer or something else in the future.  The jOOQ connection has to be open for the streaming to work so passing the consumer is the only way I figured out to make it work.

Reply:
Yeah, it makes sense and that's what callbacks are for. The javalin reference seemed odd but it makes sense to me as we have to pull the library in. Nothing wrong with using something useful.